test/headers_self_contained/tstHeader.cpp

```diff
@@ -19,4 +19,8 @@
 #include ARBORX_HEADER_TO_TEST
 #include ARBORX_HEADER_TO_TEST
 
+#if !defined(ARBORX_CONFIG_HPP)
+#error "ARBORX_HEADER_TO_TEST does not include AroborX_Config.hpp"
+#endif
```

Comment:
AroborX_ -> ArborX_

Also why would we enforce that in the 1st place?

Reply:
Otherwise, we can never be sure that the preprocessor variables defined in `ArborX_Config.hpp` are visible everywhere.

Reply:
I understand that we probably want that to be visible in places that use these macros.  I maintain it is not appropriate to include everywhere.  Definitely not in the Point or Container headers and such.

Reply:
But the question is: how do we avoid the situation where we think they are defined, but actually are not? The answer should not be "just remember".

Reply:
I don't care how you want to enforce it, I feel pretty strong against the current solution.

Reply:
Probably??? For me, these are rather hard to detect problems that can easily be avoided by this check and just including the file everywhere.

Reply:
This is NOT the only solution.  You can explicitly list the file that have the macro and pass a definition that activate the check.  This include is NOT getting into the geometry, heap, nor container headers.  It is a ridiculous solution.

Reply:
This is the same as for `Kokkos` where you need `Kokkos_Macros.hpp` practically everywhere.
For me, the content is independent of the purpose here. We need a place where we can configuration variables that are reliably accessible. At the moment, it happens that this file just contains the two `MPI` variables so this might actually be needed only in a few files. 
But as @aprokop said it is really hard to maintain if we only include it in these few files.

I feel very strongly in favor of this solution.

Reply:
Then come up with a different solution.  Parse the config file to get the list of macro and grep for it in header at configure time to see if you should enable that check or nor.